### ISSUE TITLE
Wrinkle maps, curves/particles/jiggle/something animatable hair, faceit compatibility/instructions

### ISSUE THREAD

**Issue by Vidyut** [makehumancommunity/mpfb2#126]
I was playing about with the new shader and it looks... complex. I liked the effect, but ended up deleting components I didn't need to keep things light. On the other hand, I wouldn't mind a wrinkles setup. It is complicated, but shouldn't be as complicated as our current shader and can take the character to new levels. Basic default options could be a wrinkles map layer for the skins and procedural wrinkles for key areas like forehead and eyes - you're already doing it for lips - might as well add some there and make them animatable as per surface tension :D 

Adding the Mixamo rig is brilliant, but the utility is slightly marred by not being able to animate a face. A possibility for a separate face rig that can be added would be great. Rigify has one, of course, but then we could do with a mixamo retargeter....

I just can't get the hair to come to life, so using it in animations isn't working for me. I can add particle or curves hair, but it won't play well with the rest. Can't save, so have to recreate every time... if there is a method, a good tutorial might put me out of my misery. I'm great at following tutorials. Some way to create/store particles/curves would be great. Alternatively, a way to quickly attach them each time I need.

Also I'm new to blender and mostly a dabbler (I originally planned to make a promo for my scifi book, but couldn't get the characters working to my taste fast enough. The features are brilliant and I'll keep trying, but they seem to stop just short of what seems like production quality in my head. (anybody wanting to lend a hand will have my eternal gratitude). Some of these things may be present and I may be too dumb to understand.

**Comment by Vidyut** [comment]
Update: The rigify weirdness seems mostly resolved now. I think I may have too many versions of MPFB in play over time. I deleted everything, started from scratch and it works fine but I still have problems with the files I'd made for my promo, so I guess I'll have to do it over. Planning to launch in a few days, so I guess no promo for me. I swear I'm better at writing scifi than this stuff...

**Comment by Vidyut** [comment]
N/SFW version of the skin might be good too. If we aren't showing breasts and navels and genitals, I don't know why the code can't be leaner without creating any details for them. It is a fairly heavy and unwieldy shader. It is worth stripping out whatever is not needed.

This is not to be prudish. I have no issues with skin textures or topology with genitals. But the v2 shader puts quite a hit on performance. I imagine the vast majority of uses will not need NSFW content.

**Comment by joepal1976** [comment]
I realize now that the naming of the different shaders might have been a mistake. A "v2" shader might give the indication that it somehow is a superior version compared to "v0" and that it in every way will solve every use case better. 

For performance and animation, the v2 shader is not very suitable. In these cases, you would probably want to bake the procedural aspects into texture maps and use a v0 ("makeskin") shader. 

The use case for the v2 shader is when you want to fake detail procedurally rather than have fixed texture maps. The idea with it is to put extra everything into it so that you don't have to add such details manually. The obvious downside is that it becomes heavy and complex.

Now, my idea is to provide several different shaders depending on use case. For example I'm planning on adding a cartoon shader at some point. Having different variations of the procedural shader should not be too tricky either. However, the v0/v1/v2 naming will then no longer be particularly appropriate.

**Comment by Vidyut** [comment]
Yes, I guessed I'd have to bake it out, but even for baking, etc, for users using only CPU for eg, a light version of the shader might help. That way we can use it and keep tweaking till we need to fix the look and the scene still works without having to keep baking. Vast majority of use cases seem to be clothed, so it would amount to a considerable percentage of users.

**Comment by ashledombos** [comment]
Hair part duplicates #80 (closed, haireditor now available).
